R/helpers.R

```diff
@@ -54,28 +49,46 @@ getRMNDGDP <- function(scenario="gdp_SSP2",
 #' @param isocol, name of the column containing ISO3 codes, default is "iso".
 #' @param valuecol, name of the column containing the GDP values, default is "weight".
 #' @param usecache, store the result in a RDS file in the working directory, default is FALSE.
-#' @param gdpfile, if caching is required, specify the filename here, default is "GDPcache.rds"
+#' @param isolev, if the data provided has to be on ISO or regional level
 #' @keywords gdp
 #' @import data.table
 #' @importFrom madrat calcOutput
 #' @export
 
-getRMNDGDPcap <- function(scenario="gdp_SSP2",
-                          yearcol="year",
-                          isocol="iso",
-                          valuecol="weight",
-                          usecache=F,
-                          gdpfile="GDPcache.rds"){
+getRMNDGDP <- function(scenario="gdp_SSP2",
+                       yearcol="year",
+                       isocol,
+                       valuecol="weight",
+                       isolev = F,
+                       usecache = F){
+  variable <- Year <- NULL
 
-  `.` <- iso <- value <- GDP_cap <- weight <- POP_val <- NULL
-  scenario <- gsub("gdp_", "", scenario)
-  gdp <- getRMNDGDP(paste0("gdp_", scenario), usecache=usecache)
-  POP_country=calcOutput("Population", aggregate = F)[,, paste0("pop_", scenario)]
-  POP <- magpie2dt(POP_country, regioncol = "iso",
-                   yearcol = "year", datacols = "POP")
-  POP=POP[,.(iso,year,POP,POP_val=value)]
-  GDP_POP=merge(gdp,POP,all = TRUE,by=c("iso","year"))
-  GDP_POP[,GDP_cap:=weight/POP_val]
+  if (isolev) {
+    gdpfile = "GDPcacheISO.rds"
```

Comment:
it is not good style to have generated files (especially files that are just dropped into the working directory) of which the user can not choose the filename.

Reply:
a possiblity would be to keep the gdpfile parameter and just append "ISO" in case the iso version needs to be stored. This should be documented, however. Or just have two cachefile parameter, that's maybe even cleaner (`gdpfile_iso, gdpfile`)

Reply:
At the moment, GDP is required in both ISO and regional level in different parts of the code, hence the split into 2 functions. They can be made cleaner, I will clean them up and allow for 2 caches one for each regional aggregation.